packages/task/src/browser/task-configurations.ts

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { inject, injectable } from 'inversify';
+import { TaskOptions } from '../common/task-protocol';
+import { ILogger, Disposable, DisposableCollection } from '@theia/core/lib/common/';
+import URI from "@theia/core/lib/common/uri";
+import { FileSystemWatcherServer, DidFilesChangedParams, FileChange } from '@theia/filesystem/lib/common/filesystem-watcher-protocol';
+import { FileSystem } from '@theia/filesystem/lib/common';
+import * as jsoncparser from 'jsonc-parser';
+import { ParseError } from 'jsonc-parser';
+
+export interface TaskConfigurationClient {
+    /**
+     * A task has been added or removed from the known task configurations
+     * @returns an array of strings, each one being a task label
+     */
+    taskConfigurationChanged: (event: string[]) => void;
+}
+
+/**
+ * Watches potentially two tasks.json files: one in the user's home directory (in $home/.theia/ ),
+ * and  another one in the current workspace ( in $workspace/.theia/ )
+ */
+@injectable()
+export class TaskConfigurations implements Disposable {
+
+    /* Task client to send notifications to.  */
+    protected readonly toDispose = new DisposableCollection();
+
+    // Map where all the parsed tasks, from tasks.json files, are saved
+    protected tasksMap = new Map<string, TaskOptions>();
+
+    // the tasks.json config file(s) we are watching
+    protected watchedConfigFileUris: string[] = [];
```

Comment:
so here, for instance, I don't think there is a need for that extra comment as the name already says the same. Btw. is there a reason for using different comments (SL, ML) here and there?

Reply:
Agreed, comment removed, as well as another one just above that is a similar case. 

Here, does SL, ML stand for Single Line / Multiple line? I should be more consistent, avoid mixing these together, like in this area of this file. Fixed at least here.

Reply:
Yes, I meant Single vs. Multi-line comments.
For such documentation we should use ML. For comments within bodies SL.